src/Surfnet/ServiceProviderDashboard/Application/ViewObject/EntityActions.php

```diff
@@ -167,6 +167,15 @@ public function allowSecretResetAction()
         return $meetsProtocolRequirement && $meetsPublicationStatusRequirement;
     }
 
+    public function allowChangeRequestAction(): bool
+    {
+        if ($this->readOnly) {
+            return false;
+        }
+        return $this->status == Constants::STATE_PUBLISHED;
+    }
```

Comment:
I believe we have a thorough test coverage of the neighboring methods. Please cover this one in test too.

Reply:
That is correct. This one is covered in test_change_request_action_is_allowed